examples/todo-intermediate/src/controllers/todo-list.controller.ts

```diff
@@ -0,0 +1,67 @@
+import {repository, HasManyRepository} from '@loopback/repository';
+import {TodoListRepository} from '../repositories';
+import {TodoList, Todo} from '../models';
+import {requestBody, param, post, get, del, put, patch} from '@loopback/rest';
+
+export class TodoListController {
+  protected todoRepo: HasManyRepository<TodoList>;
+
+  constructor(
+    @repository(TodoListRepository) protected todoListRepo: TodoListRepository,
+  ) {}
+
+  @post('/todo-lists')
+  async createTodoList() {
+    const newTodoList = {createdAt: new Date()};
+    return await this.todoListRepo.create(newTodoList);
+  }
+
+  @get('/todo-lists/{id}')
+  async findTodoListById(@param.path.number('id') id: number) {
+    return await this.todoListRepo.findById(id);
+  }
+
+  @get('/todo-lists')
+  async findTodos() {
+    return await this.todoListRepo.find();
+  }
+
+  @put('/todo-lists/{id}')
+  async replaceTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todoList: TodoList,
+  ) {
+    return await this.todoListRepo.replaceById(id, todoList);
+  }
+
+  @patch('/todo-lists/{id}')
+  async updateTodo(
+    @param.path.number('id') id: number,
+    @requestBody() todoList: TodoList,
+  ) {
+    return await this.todoListRepo.updateById(id, todoList);
+  }
+
+  @del('/todo-lists/{id}')
+  async deleteTodo(@param.path.number('id') id: number) {
+    return await this.todoListRepo.deleteById(id);
```

Comment:
I wonder when deleting a TodoList whether we should be deleting its Todos as well. With the context of the app being created here, this makes sense to me. Any thoughts @strongloop/sq-lb-apex?

Reply:
I think when deleting a ToDoList, it makes sense to delete the ToDos inside the list as well.

Reply:
In a typical HasMany/BelongsTo relation, the SQL database is configured with a foreign key constraint to ensure each Todo belongs to an existing TodoList. As a result, you cannot delete a TodoList before all owned Todos are deleted first, because otherwise the foreign key constraint would be violated.

Better ORMs support cascading deletes, where the SQL database is configured to automatically delete all owned Todo items when a TodoList is going to be deleted. LoopBack's autoupdate/automigrate does not support that functionality yet and NoSQL databases don't implement cascading deletes, therefore I think it's best to explicitly delete all linked Todos before deleting the TodoList they belong to. 

IMO, the implementation of cascading delete DOES NOT belong to controllers, it should be implemented at the repository level.

Reply:
Has this issue been tracked anywhere? If not, I'd like to track it and have it be separated from this issue.

I'd also like to argue that this should be implemented in the upcoming juggler instead of the repository to keep the SQL/noSQL cases separate from our repositories